ddtrace/profiling/exporter/pprof.pyx

```diff
@@ -155,6 +152,15 @@ def _get_thread_name(thread_id: typing.Optional[int], thread_name: typing.Option
     return thread_name
 
 
+cdef groupby(object collection, object key):
```

Comment:
can you be more specific with the typing here?

`list collection, str key` ?

would save Cython some type checking/conversions I think

same with return type? although the default is `object` right?

Reply:
Yep, `key` is a callable, and I'm not sure `collection` is always a list. I believe in some cases we pass in an iterable. The return type is also an iterable, so I don't know if we could do any better than `object`?

Reply:
no probably not.

Cython annotate might give some hunts as to what Cython tries to do, but... 🤷🏻